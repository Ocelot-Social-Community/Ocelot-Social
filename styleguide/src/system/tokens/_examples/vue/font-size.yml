#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
