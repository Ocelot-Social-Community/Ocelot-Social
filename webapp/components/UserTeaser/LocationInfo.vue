<template>
  <div class="location-info">
    <div class="location">
      <base-icon name="map-marker" />
      {{ location.name }}
    </div>
    <div v-if="distance" class="distance">{{ distance }}</div>
  </div>
</template>

<script>
export default {
  name: 'LocationInfo',
  props: {
    location: { type: Object, default: null },
  },
  computed: {
    distance() {
      return !this.location.distance
        ? null
        : this.$t('location.distance', { distance: this.location.distance })
    },
  },
}
</script>

<style scoped>
.location-info {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  .location {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .distance {
    margin-top: 8px;
  }
}
</style>
