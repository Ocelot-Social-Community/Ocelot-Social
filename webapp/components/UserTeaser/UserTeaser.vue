<template>
  <div class="user-teaser" v-if="displayAnonymous">
    <profile-avatar v-if="showAvatar" size="small" />
    <span class="info anonymous">{{ $t('profile.userAnonym') }}</span>
  </div>
  <div v-else :class="[{ 'disabled-content': user.disabled }]" placement="top-start">
    <div :class="['user-teaser']">
      <nuxt-link v-if="linkToProfile && showAvatar" :to="userLink" data-test="avatarUserLink">
        <profile-avatar :profile="user" size="small" />
      </nuxt-link>
      <profile-avatar v-else-if="showAvatar" :profile="user" size="small" />
      <div class="info flex-direction-column">
        <div :class="wide ? 'flex-direction-row' : 'flex-direction-column'">
          <nuxt-link v-if="linkToProfile" :to="userLink">
            <span class="text">
              <span class="slug">{{ userSlug }}</span>
              <span v-if="!userOnly" class="name">{{ userName }}</span>
            </span>
          </nuxt-link>
          <span v-else class="text">
            <span class="slug">{{ userSlug }}</span>
            <span v-if="!userOnly" class="name">{{ userName }}</span>
          </span>
          <span v-if="wide">&nbsp;</span>
          <span v-if="group">
            <span class="text">
              {{ $t('group.in') }}
            </span>
            <nuxt-link :to="groupLink">
              <span class="text">
                <span class="slug">{{ groupSlug }}</span>
                <span v-if="!userOnly" class="name">{{ groupName }}</span>
              </span>
            </nuxt-link>
          </span>
        </div>
        <span v-if="!userOnly" class="text">
          <base-icon name="clock" />
          <relative-date-time :date-time="dateTime" />
          <slot name="dateTime"></slot>
        </span>
        <span v-else class="text">{{ userName }}</span>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'

import RelativeDateTime from '~/components/RelativeDateTime'
import ProfileAvatar from '~/components/_new/generic/ProfileAvatar/ProfileAvatar'

export default {
  name: 'UserTeaser',
  components: {
    RelativeDateTime,
    ProfileAvatar,
  },
  props: {
    linkToProfile: { type: Boolean, default: true },
    user: { type: Object, default: null },
    group: { type: Object, default: null },
    wide: { type: Boolean, default: false },
    showAvatar: { type: Boolean, default: true },
    dateTime: { type: [Date, String], default: null },
    showPopover: { type: Boolean, default: true },
  },
  computed: {
    ...mapGetters({
      isModerator: 'auth/isModerator',
    }),
    itsMe() {
      return this.user.slug === this.$store.getters['auth/user'].slug
    },
    displayAnonymous() {
      const { user, isModerator } = this
      return !user || user.deleted || (user.disabled && !isModerator)
    },
    userLink() {
      const { id, slug } = this.user
      if (!(id && slug)) return ''
      return { name: 'profile-id-slug', params: { slug, id } }
    },
    userSlug() {
      const { slug } = this.user || {}
      return slug && `@${slug}`
    },
    userName() {
      const { name } = this.user || {}
      return name || this.$t('profile.userAnonym')
    },
    userOnly() {
      return !this.dateTime && !this.group
    },
    groupLink() {
      const { id, slug } = this.group
      if (!(id && slug)) return ''
      return { name: 'groups-id-slug', params: { slug, id } }
    },
    groupSlug() {
      const { slug } = this.group || {}
      return slug && `&${slug}`
    },
    groupName() {
      const { name } = this.group || {}
      return name || this.$t('profile.userAnonym')
    },
  },
  methods: {
    optimisticFollow({ followedByCurrentUser }) {
      const inc = followedByCurrentUser ? 1 : -1
      this.user.followedByCurrentUser = followedByCurrentUser
      this.user.followedByCount += inc
    },
    updateFollow({ followedByCurrentUser, followedByCount }) {
      this.user.followedByCount = followedByCount
      this.user.followedByCurrentUser = followedByCurrentUser
    },
  },
}
</script>

<style lang="scss">
.trigger {
  max-width: 100%;
}

.user-teaser {
  display: flex;
  flex-wrap: nowrap;

  > .profile-avatar {
    flex-shrink: 0;
  }

  > .info {
    padding-left: $space-xx-small;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;

    &.anonymous {
      font-size: $font-size-base;
    }

    .slug {
      color: $color-primary;
      font-size: $font-size-base;
    }

    .name {
      color: $text-color-soft;
      font-size: $font-size-small;
    }
  }

  .flex-direction-column {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  .flex-direction-row {
    display: flex;
    flex-direction: row;
    justify-content: center;
  }

  .text {
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;

    > .ds-text {
      display: inline;
    }
  }
}
</style>
