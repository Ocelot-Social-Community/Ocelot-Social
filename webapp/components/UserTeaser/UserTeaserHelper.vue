<template>
  <span v-if="!linkToProfile">
    <slot />
  </span>
  <button v-else-if="linkToProfile && showPopover && isTouchDevice" @click.prevent="openMenu">
    <slot />
  </button>
  <nuxt-link
    v-else
    :to="userLink"
    @mouseover.native="() => showPopover && openMenu()"
    @mouseleave.native="closeMenu"
  >
    <slot />
  </nuxt-link>
</template>

<script>
export default {
  name: 'UserTeaserHelper',
  props: {
    userLink: { type: Object },
    linkToProfile: { type: Boolean, default: true },
    showPopover: { type: Boolean, default: false },
    isTouchDevice: { type: Boolean, default: false },
  },
  methods: {
    openMenu() {
      this.$emit('open-menu')
    },
    closeMenu() {
      this.$emit('close-menu')
    },
  },
}
</script>
