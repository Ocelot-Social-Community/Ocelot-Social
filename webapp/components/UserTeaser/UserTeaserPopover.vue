<template>
  <div class="user-teaser-popover">
    <badges
      v-if="$env.BADGES_ENABLED"
      :badges="[user.badgeVerification, ...user.badgeTrophiesSelected]"
    />
    <location-info
      v-if="user.location"
      :location="{ ...user.location, distanceToMe: user.distanceToMe }"
      class="location-info"
    />
    <ul class="statistics">
      <li>
        <ds-number :count="user.followedByCount" :label="$t('profile.followers')" />
      </li>
      <li>
        <ds-number
          :count="user.contributionsCount"
          :label="$t('common.post', null, user.contributionsCount)"
        />
      </li>
      <li>
        <ds-number
          :count="user.commentedCount"
          :label="$t('common.comment', null, user.commentedCount)"
        />
      </li>
    </ul>
    <nuxt-link v-if="isTouchDevice && userLink" :to="userLink" class="link">
      <ds-button primary>{{ $t('user-teaser.popover.open-profile') }}</ds-button>
    </nuxt-link>
  </div>
</template>

<script>
import Badges from '~/components/Badges.vue'
import LocationInfo from '~/components/UserTeaser/LocationInfo.vue'
import { isTouchDevice } from '~/components/utils/isTouchDevice'

export default {
  name: 'UserTeaserPopover',
  components: {
    Badges,
    LocationInfo,
  },
  props: {
    user: { type: Object, default: null },
    userLink: { type: Object },
  },
  computed: {
    isTouchDevice() {
      return isTouchDevice()
    },
  },
}
</script>

<style scoped>
.user-teaser-popover {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 16px;
  gap: 16px;
  min-width: 200px;
}

.location-info {
  margin-bottom: 12px;
}

.link {
  margin-top: 16px;
}

.statistics {
  display: flex;
  justify-content: space-between;
  width: 100%;
}
</style>
