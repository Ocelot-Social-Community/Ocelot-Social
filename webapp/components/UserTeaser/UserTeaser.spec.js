import { render, screen, fireEvent } from '@testing-library/vue'
import { RouterLinkStub } from '@vue/test-utils'
import UserTeaser from './UserTeaser.vue'
import Vuex from 'vuex'

const localVue = global.localVue

/*
to test:
  - user is deleted
  - user is disabled
  - user is anonymous
  - user is a moderator
  - user is a moderator and the user is deleted
  - user is a moderator and the user is disabled
  - linkToProfile on/off
  - showAvatar on/off
  - showPopover on/off
  - hovering name on desktop
  - clicking name on mobile
  - hovering avatar on desktop
  - clicking avatar on mobile
  - hovering slug on desktop
  - clicking slug on mobile
*/

// Mock Math.random, used in Dropdown
Object.assign(Math, {
  random: () => 0,
})

const userTilda = {
  name: 'Tilda Swinton',
  slug: 'tilda-swinton',
  id: 'user1',
  badgeVerification: {
    id: 'bv1',
    icon: '/icons/verified',
    description: 'Verified',
    isDefault: false,
  },
  badgeTrophiesSelected: [
    {
      id: 'trophy1',
      icon: '/icons/trophy1',
      description: 'Trophy 1',
      isDefault: false,
    },
    {
      id: 'trophy2',
      icon: '/icons/trophy2',
      description: 'Trophy 2',
      isDefault: false,
    },
    {
      id: 'empty',
      icon: '/icons/empty',
      description: 'Empty',
      isDefault: true,
    },
  ],
}

describe('UserTeaser', () => {
  const Wrapper = ({
    isModerator = false,
    withLinkToProfile = true,
    onTouchScreen = false,
    user = userTilda,
  }) => {
    const mockIsTouchDevice = onTouchScreen
    jest.mock('../utils/isTouchDevice', () => ({
      isTouchDevice: jest.fn(() => mockIsTouchDevice),
    }))
    const store = new Vuex.Store({
      getters: {
        'auth/user': () => {
          return {}
        },
        'auth/isModerator': () => isModerator,
      },
    })
    return render(UserTeaser, {
      localVue,
      store,
      propsData: {
        user,
        linkToProfile: withLinkToProfile,
      },
      stubs: {
        NuxtLink: RouterLinkStub,
        'user-teaser-popover': true,
        'v-popover': true,
        'client-only': true,
      },
      mocks: {
        $t: jest.fn((t) => t),
      },
    })
  }

  it('renders anonymous user', () => {
    const wrapper = Wrapper({ user: null })
    expect(wrapper.container).toMatchSnapshot()
  })

  describe('given an user', () => {
    describe('without linkToProfile, on touch screen', () => {
      let wrapper
      beforeEach(() => {
        wrapper = Wrapper({ withLinkToProfile: false, onTouchScreen: true })
      })

      it('renders', () => {
        expect(wrapper.container).toMatchSnapshot()
      })

      describe('when clicking the user name', () => {
        beforeEach(async () => {
          const userName = screen.getByText('Tilda Swinton')
          await fireEvent.click(userName)
        })

        it('renders the popover', () => {
          expect(wrapper.container).toMatchSnapshot()
        })
      })
    })

    describe('with linkToProfile, on touch screen', () => {
      let wrapper
      beforeEach(() => {
        wrapper = Wrapper({ withLinkToProfile: true, onTouchScreen: true })
      })

      it('renders', () => {
        expect(wrapper.container).toMatchSnapshot()
      })

      describe('when clicking the user name', () => {
        beforeEach(async () => {
          const userName = screen.getByText('Tilda Swinton')
          await fireEvent.click(userName)
        })

        it('renders the popover', () => {
          expect(wrapper.container).toMatchSnapshot()
        })
      })
    })

    describe('without linkToProfile, on desktop', () => {
      let wrapper
      beforeEach(() => {
        wrapper = Wrapper({ withLinkToProfile: false, onTouchScreen: false })
      })

      it('renders', () => {
        expect(wrapper.container).toMatchSnapshot()
      })

      describe('when hovering the user name', () => {
        beforeEach(async () => {
          const userName = screen.getByText('Tilda Swinton')
          await fireEvent.mouseOver(userName)
          await new Promise((resolve) => setTimeout(resolve, 2000))
        })

        it('renders the popover', () => {
          expect(wrapper.container).toMatchSnapshot()
        })
      })
    })

    describe('with linkToProfile, on desktop', () => {
      let wrapper
      beforeEach(() => {
        wrapper = Wrapper({ withLinkToProfile: true, onTouchScreen: false })
      })

      it('renders', () => {
        expect(wrapper.container).toMatchSnapshot()
      })

      describe('when hovering the user name', () => {
        beforeEach(async () => {
          const userName = screen.getByText('Tilda Swinton')
          await fireEvent.mouseOver(userName)
          await new Promise((resolve) => setTimeout(resolve, 2000))
        })

        it('renders the popover', () => {
          expect(wrapper.container).toMatchSnapshot()
        })
      })
    })

    describe('user is deleted', () => {
      it('renders anonymous user', () => {
        const wrapper = Wrapper({ user: { ...userTilda, deleted: true } })
        expect(wrapper.container).toMatchSnapshot()
      })

      describe('even if the current user is a moderator', () => {
        it('renders anonymous user', () => {
          const wrapper = Wrapper({
            user: { ...userTilda, deleted: true },
            isModerator: true,
          })
          expect(wrapper.container).toMatchSnapshot()
        })
      })
    })

    describe('user is disabled', () => {
      it('renders anonymous user', () => {
        const wrapper = Wrapper({ user: { ...userTilda, disabled: true } })
        expect(wrapper.container).toMatchSnapshot()
      })

      describe('current user is a moderator', () => {
        it('renders user name', () => {
          const wrapper = Wrapper({ user: { ...userTilda, disabled: true }, isModerator: true })
          expect(wrapper.container).toMatchSnapshot()
        })
      })
    })
  })
})
