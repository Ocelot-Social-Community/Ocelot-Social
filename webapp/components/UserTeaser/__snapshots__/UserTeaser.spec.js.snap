// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTeaser given an user avatar is disabled does not render the avatar 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <!---->
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <a>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </a>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div />
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user user is deleted even if the current user is a moderator renders anonymous user 1`] = `
<div>
  <div
    class="user-teaser"
  >
    <div
      class="profile-avatar --small --no-image"
    >
      <span
        class="initials"
      >
        
      </span>
       
      <span
        class="base-icon"
      >
        <!---->
      </span>
       
      <!---->
    </div>
     
    <span
      class="info anonymous"
    >
      profile.userAnonym
    </span>
  </div>
</div>
`;

exports[`UserTeaser given an user user is deleted renders anonymous user 1`] = `
<div>
  <div
    class="user-teaser"
  >
    <div
      class="profile-avatar --small --no-image"
    >
      <span
        class="initials"
      >
        
      </span>
       
      <span
        class="base-icon"
      >
        <!---->
      </span>
       
      <!---->
    </div>
     
    <span
      class="info anonymous"
    >
      profile.userAnonym
    </span>
  </div>
</div>
`;

exports[`UserTeaser given an user user is disabled current user is a moderator renders user name 1`] = `
<div>
  <div
    class="disabled-content"
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <a>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </a>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <a>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </a>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div />
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user user is disabled renders anonymous user 1`] = `
<div>
  <div
    class="user-teaser"
  >
    <div
      class="profile-avatar --small --no-image"
    >
      <span
        class="initials"
      >
        
      </span>
       
      <span
        class="base-icon"
      >
        <!---->
      </span>
       
      <!---->
    </div>
     
    <span
      class="info anonymous"
    >
      profile.userAnonym
    </span>
  </div>
</div>
`;

exports[`UserTeaser given an user with linkToProfile, on desktop renders 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <a>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </a>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <a>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </a>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div />
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user with linkToProfile, on desktop when hovering the user name renders the popover 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      open="true"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <a>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </a>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <a>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </a>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div>
        <user-teaser-popover-stub
          link-to-profile="true"
          user="[object Object]"
          userlink="[object Object]"
        />
      </div>
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user with linkToProfile, on touch screen renders 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <a>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </a>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <a>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </a>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div />
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user with linkToProfile, on touch screen when clicking the user name renders the popover 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <a>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </a>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <a>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </a>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div />
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user without linkToProfile, on desktop renders 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <span>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </span>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <span>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </span>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div />
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user without linkToProfile, on desktop when hovering the user avatar renders the popover 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      open="true"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <span>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </span>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <span>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </span>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div>
        <user-teaser-popover-stub
          user="[object Object]"
          userlink="[object Object]"
        />
      </div>
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user without linkToProfile, on desktop when hovering the user name renders the popover 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      open="true"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <span>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </span>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <span>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </span>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div>
        <user-teaser-popover-stub
          user="[object Object]"
          userlink="[object Object]"
        />
      </div>
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user without linkToProfile, on touch screen renders 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <span>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </span>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <span>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </span>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div />
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user without linkToProfile, on touch screen when clicking the user avatar renders the popover 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      open="true"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <span>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </span>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <span>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </span>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div>
        <user-teaser-popover-stub
          user="[object Object]"
          userlink="[object Object]"
        />
      </div>
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser given an user without linkToProfile, on touch screen when clicking the user name renders the popover 1`] = `
<div>
  <div
    class=""
    placement="top-start"
  >
    <v-popover-stub
      autohide="true"
      class="user-teaser"
      container="body"
      delay="0"
      handleresize="true"
      offset="16"
      open="true"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <span>
        <div
          class="profile-avatar --small"
        >
          <span
            class="initials"
          >
            TS
          </span>
           
          <!---->
           
          <img
            alt="Tilda Swinton"
            class="image"
            src="/api/avatars/tilda-swinton"
            title="Tilda Swinton"
          />
        </div>
      </span>
       
      <div
        class="info flex-direction-column"
      >
        <div
          class="flex-direction-column"
        >
          <span>
            <span
              class="text"
            >
              <span
                class="slug"
              >
                @tilda-swinton
              </span>
               
              <span
                class="name"
              >
                Tilda Swinton
              </span>
            </span>
          </span>
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div>
        <user-teaser-popover-stub
          user="[object Object]"
          userlink="[object Object]"
        />
      </div>
    </v-popover-stub>
  </div>
</div>
`;

exports[`UserTeaser renders anonymous user 1`] = `
<div>
  <div
    class="user-teaser"
  >
    <div
      class="profile-avatar --small --no-image"
    >
      <span
        class="initials"
      >
        
      </span>
       
      <span
        class="base-icon"
      >
        <!---->
      </span>
       
      <!---->
    </div>
     
    <span
      class="info anonymous"
    >
      profile.userAnonym
    </span>
  </div>
</div>
`;
